src/box/lua/console.lua

```diff
@@ -611,7 +613,11 @@ local function local_read(self)
             if not lang or lang == 'lua' then
                 -- stop once a complete Lua statement is entered
                 if local_check_lua(buf) then
-                    break
+                    if not line:endswith(multiline_delim) then
```

Comment:
The main motivation here is to allow to copy-paste tests into console. When I pasted several lines with backslash escaped newlines at once, the backlashes were not stripped.

Reply:
It has something to do with the way copy-paste works in linux. Python shell works the same way:
```
Python 3.9.13 (main, May 18 2022, 00:00:00) 
[GCC 11.3.1 20220421 (Red Hat 11.3.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = '''a = 10\ 
... + 12\ 
... + 42
... '''
>>> a
'a = 10\\ \n+ 12\\ \n+ 42\n'
>>> a = 10\ 
  File "<stdin>", line 1
    a = 10\ 
           ^
SyntaxError: unexpected character after line continuation character
>>> + 12\ 
  File "<stdin>", line 1
    + 12\ 
         ^
SyntaxError: unexpected character after line continuation character
>>> + 42
42
>>> a = 10 \
... + 12 \
... + 42
>>> a
64
>>> 
```
The text without a space symbol before backslash+newline get a space (ASCII 32) added between them in each line while copy-pasting. At the same time, strings that have a space char before backslash+newline are handled correctly. Thus the patch is correct:
```
Tarantool 2.10.0-beta1-951-gba62be705
type 'help' for interactive help
tarantool> a = 10 \
         > + 12 \
         > + 42
---
...

tarantool> a = 10\ 
---
- error: '[string "a = 10\ "]:1: unexpected symbol near ''\'''
...

tarantool> + 12\ 
---
- error: '[string "+ 12\ "]:1: unexpected symbol near ''+'''
...

tarantool> + 42
---
- error: '[string "+ 42"]:1: unexpected symbol near ''+'''
...
```

Reply:
Oh, I was quite wrong. There is nothing wrong with copy-paste mechanic. Vim for somewhat reason (alignment?) places space symbols at the end of such lines. When you use cat, it's all fine.

Reply:
The problem is in readline8.1, as it makes bracketed paste mode default. Fixed in intermediate commit.